toolchains/solidity/extension/src/gas-estimation.ts

```diff
@@ -0,0 +1,344 @@
+import { execSync, exec } from "child_process";
+import * as vscode from "vscode";
+
+type GasReport = {
+  average: bigint;
+  min?: bigint;
+  max?: bigint;
+  median?: bigint;
+};
+
+type Function = {
+  name: string,
+  line: number
+};
+
+type Report = Map<string, Map<string, GasReport>>;
+
+type ReportDecorators = Map<string, vscode.DecorationOptions[]>;
+
+function isForgeInstalled(): boolean {
+  try {
+    execSync("forge --version");
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+async function gasReportTests(cwd: string): Promise<ReportDecorators> {
+  const reports: Map<string, Report> = new Map();
+  let decorations: ReportDecorators = new Map();
+
+  if (!isForgeInstalled()) {
+    console.log("Forge is not installed");
+    return decorations;
+  }
+
+
+  // Gas estimation from the tests
+  await new Promise<void>((resolve, reject) => exec("forge test --gas-report", { cwd }, async (error: any, _stdout: any, _stderr: any) => {
+    if (error) {
+      console.log("error", error);
+      reject(error);
+    }
+
+     if (_stdout === "null") {
+        resolve();
+      }
+
+    let contractName = "";
+    await Promise.all(_stdout.split("\n").map(async (line: string) => {
+      const lineParts = line.split("|");
+      if (lineParts.length === 8) {
+        const trimmedLineParts = lineParts.map((part) => part.trim());
+        if (trimmedLineParts[1] !== "" && trimmedLineParts[2] === "" && trimmedLineParts[3] === "" && trimmedLineParts[4] === "" && trimmedLineParts[5] === "" && trimmedLineParts[6] === "") {
```

Comment:
If the check is so verbose, why not verify all the fields ?

Reply:
The last fields are really useless as they always are just a single space